application-ldapuserimport-api/src/main/java/com/xwiki/ldapuserimport/script/LDAPUserImportScriptService.java

```diff
@@ -193,4 +203,18 @@ public LDAPUserImportConfiguration getConfiguration()
     {
         return configuration;
     }
+
+    /**
+     * Starts an LDAP Group import job.
+     *
+     * @return the ongoing LDAP Group import job
+     * @throws Exception in case of exceptions
+     * @since 1.4
+     */
+    @Unstable
+    public AbstractLDAPGroupImportJob importLDAPGroups() throws Exception
```

Comment:
```suggestion
    public Job importLDAPGroups()
```

I don't think you need to expose the Job implementation and script services are not supposed to throw exceptions normally. The practice is to catch the exception and add a ``getLastError()`` method. See for instance https://github.com/xwiki/xwiki-platform/blob/ec20f5492bd229203414ec34211600acc5bf9a0e/xwiki-platform-core/xwiki-platform-component/xwiki-platform-component-script/src/main/java/org/xwiki/component/script/ComponentScriptService.java#L190 .

Reply:
To my knowledge, we should now throw exceptions in script services since the `#try` directive has been added to Velocity ; see https://extensions.xwiki.org/xwiki/bin/view/Extension/Script%20Module/#HBestPractices.

Regarding exposing the job implementation, I'll remove it according to the discussion on https://github.com/xwikisas/application-ldapuserimport/pull/67#discussion_r1211497082

Reply:
Fixed job implementation in 377d72c